thorlcr/slave/traceslave.hpp

```diff
@@ -173,15 +173,15 @@ class CTracingStream : public CSimpleInterfaceOf<IEngineRowStream>, implements I
         // NOTE - cannot be called by more than one thread
         buffers[1].init(traceQueueSize+1);
         rowBufferLogCache.init(traceQueueSize);
-        int bufToDump = atomic_read(&rowBufInUse);
+        int bufToDump = rowBufInUse;
         int inactiveBuf = 1 - bufToDump;
 
         // Queue any remaining rows from inactive buffer as the oldest row is skipped
         // when queue out from active buffer to avoid race condition issues (this inactive buffer
         // may have been formerly an active buffer)
         buffers[inactiveBuf].queueOut(rowBufferLogCache, false);
 
-        atomic_set(&rowBufInUse, inactiveBuf);// Swap Active & inactiveBuf
+        rowBufInUse = inactiveBuf;// Swap Active & inactiveBuf
```

Comment:
Not new but is this safe?

Reply:
I don't think this code is 100% thread safe.  A row could be being appended at the same time as queueOut is called.  I think that could potentially cause a row to be accessed that is then overwritten, or possibly other issues.  However that is a separate issue.